dags/automated_transformation/automation_dag.py

```diff
@@ -91,18 +96,42 @@ def discover_files(ti):
 
         config = ti.dag_run.conf.copy()
         bucket = config.get("raw_data_bucket")
-        data_prefix = config.get("raw_data_prefix")
+        raw_data_prefix = config.get("raw_data_prefix")
+        raw_data_regex = config.get("raw_data_filter_regex")
         ext = config.get("ext")  # .nc as well
-        generated_list = get_all_s3_keys(bucket, data_prefix, ext)
-        chunk_size = int(len(generated_list) / 900) + 1
-        return [
-            generated_list[i: i + chunk_size]
-            for i in range(0, len(generated_list), chunk_size)
-        ]
-
+        generated_list = get_all_s3_keys(bucket, raw_data_prefix, ext)
+        collection_name = config.get("collection_name")
+        print(f"[ TOTAL DISCOVERED : {len(generated_list)}]")
 
-    @task(max_active_tis_per_dag=1)
-    def process_files(file_url, **kwargs):
+        # Filter by raw data regex
+        pattern = rf"{raw_data_prefix}/{raw_data_regex}"
+        filtered_files = [
+            f for f in generated_list if re.match(pattern, f)
+        ]
+        print(f"[ FILTERED BY PATTERN {pattern} : {len(filtered_files)}]")
+
+        # Write this to s3
+        airflow_vars_json = Variable.get("aws_dags_variables", deserialize_json=True)
+        bucket_output = airflow_vars_json.get("EVENT_BUCKET")
+        key = f"s3://{bucket_output}/events/{collection_name}"
+        chunks_xcom = []
+        
+        if len(filtered_files) > 900:
+            # Do chunking only if there are more than 900 files
+            chunk_size = max(int(len(filtered_files) / 900), 1)
+            for i in range(0, len(filtered_files), chunk_size):
+                tmp = filtered_files[i: i + chunk_size]
+                output_key = write_xcom_to_s3(f"{key}/chunk_{i}", tmp)
+                chunks_xcom.append(output_key)
+        else:
+            # put all inside same s3 file if it is less than 900
+            output_key = write_xcom_to_s3(f"{key}/all_files", filtered_files)
+            chunks_xcom.append(output_key) 
+        return chunks_xcom
+
+
+    @task(max_active_tis_per_dag=20)
```

Comment:
We probably need to make this value configurable to limit the parallel runs, so we can prevent a full redeployment to just run less or more tasks.